### ISSUE TITLE
Remove admin role name change setting

### ISSUE THREAD

**Issue by jenlampton** [backdrop/backdrop-issues#2111]
Follow-up to https://github.com/backdrop/backdrop-issues/issues/2091#issuecomment-243737629 From @klonos: 

We should move the ability to define a custom name for anonymous users in that page, but people can already achieve that goal by simply editing the label of the in-built Anonymous role. Right? So, how about we remove this fieldset entirely?

**Comment by jenlampton** [comment]
Right now the two things are different. One is the name admins see in places like the permissions page. The other is the thing visitors see in a comment in place of the username. I don't see any reason they couldn't be the same, but it's going to be more than a UI change, `theme_username()` will need to be updated as well.

**Comment by klonos** [comment]
> Right now the two things are different. One is the name admins see in places like the permissions page. The other is the thing visitors see in a comment in place of the username.

Hmm, wasn't aware of that. Explains why we had two separate things, but seems like having a different admin label than what visitors see is a 20% case. I say lets simplify unless someone objects and has a valid use case where these need to be labeled differently.

**Comment by jenlampton** [comment]
> seems like having a different admin label than what visitors see is a 20% case

Yeah, the visitor-label is actually a very old feature, we didn't get the ability to rename "anonymous" and "authenticated" until Drupal 7.

**Comment by quicksketch** [comment]
This sounds good to me too.

**Comment by Dinornis** [comment]
Yes, for removing the Anonymous Visitors fieldset from the Roles page.

**Comment by BWPanda** [comment]
Here's a PR, hope I got them all: https://github.com/backdrop/backdrop/pull/3344

As much as I agree with this issue, I have a concern: This seems like an API change in that:

1. Anything calling `config_get('system.core', 'anonymous')` will no longer work
2. What happens if anyone was adding extra fields to the 'Anonymous visitors' fieldset?

**Comment by stpaultim** [comment]
I support the intent of this PR, but it sounds like there might be backward compatibility issues.
